### PR TITLE
Consolidate screen-sharing requests in one place

### PR DESCRIPTION
**Pull request by DavDo**
> [!NOTE]
> Sorry for the large pull request. I couldn't split these changes. 

> [!IMPORTANT]
> Please ignore tests, because I'll make sure to provide a fix for all the tests on the next PR.

**Jira issue:**
https://glia.atlassian.net/browse/MOB-3055

**What was solved?**
- Remove the old CV layer and add new simpler classes
- Remove ScreenSharingController.kt
- Move Screen-Sharing functionality to OperatorRequestController

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/CallVisualizer.java [hunk 1]
```diff
@@ -25,14 +25,28 @@ public interface CallVisualizer {
...
+
     /**
      * Shows the visitor code in a dialog box on top of the current activity within your application.
      * <p>
      * This dialog will overlay on top of any other dialogs without dismissing them.
      * <p>
      * Otherwise, it behaves the same way as {@link #createVisitorCodeView(Context)}.
      */
-    void showVisitorCodeDialog(@NonNull Context context);
+    void showVisitorCodeDialog();
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/CallVisualizer.java:49]: 👍

File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogController.kt [hunk 2]
```diff
@@ -154,9 +166,9 @@ internal class DialogController(
     override fun addCallback(callback: DialogContract.Controller.Callback) {
         Logger.d(TAG, "addCallback")
         viewCallbacks.add(callback)
-        if (viewCallbacks.size == 1) {
-            dialogManager.showNext()
-        }
+//        if (viewCallbacks.size == 1) {
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogController.kt:169]: leftovers?

File: widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java [hunk 3]
```diff
@@ -87,12 +87,19 @@ public static void onAppCreate(Application application) {
...
-                getControllerFactory().getDialogController(),
-                getControllerFactory().getActivityWatcherForCallVisualizerController()
-            );
-        application.registerActivityLifecycleCallbacks(activityWatcherForCallVisualizer);
+//        ActivityWatcherForCallVisualizer activityWatcherForCallVisualizer =
+//            new ActivityWatcherForCallVisualizer(
+//                getControllerFactory().getDialogController(),
+//                getControllerFactory().getActivityWatcherForCallVisualizerController()
+//            );
+//        application.registerActivityLifecycleCallbacks(activityWatcherForCallVisualizer);
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java:95]: Leftovers?

File: widgetssdk/src/main/java/com/glia/widgets/view/Dialogs.kt [hunk 4]
```diff
@@ -304,11 +302,11 @@ internal object Dialogs {
...
-                Dependencies.getControllerFactory().dialogController.dismissVisitorCodeDialog()
-                context.asActivity().takeIf { it is CallVisualizerSupportActivity }?.apply {
-                    overridePendingTransition(0, 0)
-                    finish()
-                }
+                Dependencies.getControllerFactory().callVisualizerController.dismissVisitorCodeDialog()
+//                context.asActivity().takeIf { it is DialogHolderActivity }?.apply { TODO check
+//                    overridePendingTransition(0, 0)
+//                    finish()
+//                }
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/java/com/glia/widgets/view/Dialogs.kt:309]: And here //

**Review by AndriiHorishniiMOC — APPROVED** [review]
Great job!

File: widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionActivityWatcher.kt [hunk 5]
```diff
@@ -26,26 +26,28 @@ internal class EngagementCompletionActivityWatcher(controller: EngagementComplet
 
     private fun handleState(activityReference: WeakReference<Activity>, event: OneTimeEvent<EngagementCompletionState>) {
         val state = event.value
-        val activity = activityReference.get() ?: return
+        val activity = activityReference.get()
 
         when {
-            event.consumed || activity.isFinishing -> Logger.d(TAG, "skipping state: $state activity: $activity")
+            event.consumed -> Logger.d(TAG, "skipping.. event: ${event.value} is already consumed")
+            //finish activities event if app is in a background
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionActivityWatcher.kt:33]: It took me a while to understand this comment. It's confusing that comment says "app is in background" but event name is "QueuingOrEngagementEnded". I think we need to explain a bit more.

**Review by andrews-moc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt [hunk 6]
```diff
@@ -176,6 +179,24 @@ internal class ChatController(
         chatState = ChatState()
         subscribeToEngagement()
         decideOnQueueingUseCase().unSafeSubscribe(::enqueueForEngagement)
+        screenSharingUseCase().unSafeSubscribe(::handleScreenSharingState)
+    }
+
+    private fun handleScreenSharingState(screenSharingState: ScreenSharingState) {
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt:185]: Can we extract repeated code?

**Review by Pelkar — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt (hunk 6 above)
Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt:185]: I don't think so.
Even though we process this state in all places inside similar `when` blocks, the functions we call based on every state are different.

**Pull request merged**